src/main/java/org/apache/commons/imaging/formats/gif/GifImageParser.java

```diff
@@ -439,6 +439,16 @@ private GifBlock findBlock(final List<GifBlock> blocks, final int code) {
         return null;
     }
 
+    private <T extends GifBlock> List<T> findAllBlocks(final List<GifBlock> blocks, final int code) {
+        final List<T> filteredBlocks = new ArrayList<>();
+        for (final GifBlock gifBlock : blocks) {
+            if (gifBlock.blockCode == code) {
+                filteredBlocks.add((T) gifBlock);
```

Comment:
Hmmm, this is tricky. We could have a malformed GIF that has a block with the wrong code, resulting in a runtime exception here I think? Or if I request something like `List<ImageDescriptor> descriptors = findAllBlocks(imageContents.blocks, GRAPHIC_CONTROL_EXTENSION);`. I wonder if we could come up with a different approach to avoid issues like this?

Reply:
(it seems to be internal only, so maybe this should be fine for now)

Reply:
Just re-read the `GifImageParser#readBlocks()`, and actually I believe the only way this could happen, is if the `switch/case` logic was wrong, and used the wrong class for a block code. So this should be fine, especially as this appears to be internal use only.

So maybe just add the suppress annotation and a note, e.g.

```
// See GifImageParser#parse for reference how the blocks are created. They should match
// the code we are giving here, returning the correct class type. Internal only.
@SuppressWarnings("unchecked")
private <T extends GifBlock> List<T> findAllBlocks(final List<GifBlock> blocks, final int code) {
```

?

Reply:
I think that sounds good. I'll add the comment and suppress the warning.